### PR TITLE
ensure `hidePrevented.bs.modal` can be prevented

### LINKED ISSUE #31615: Bug in modal.js: defaultPrevented called on jQuery.Event instead of isDefaultPrevented()
Tried to prevent the `hidePrevented.bs.modal` event using `e.preventDefault()`, but it won't work since the code checks for `defaultPrevented` instead of `isDefaultPrevented()`. 

The bug on this line: https://github.com/twbs/bootstrap/blob/v4.5.2/js/src/modal.js#L237

All the other events are properly using `isDefaultPrevented()` see for example this line: https://github.com/twbs/bootstrap/blob/v4.5.2/js/src/modal.js#L117

Here's a simple fix: https://github.com/twbs/bootstrap/compare/v4.5.2...joakimriedel:joakimriedel/defaultPrevented?expand=1

Patch for 4.5.2 ?

### PR DESCRIPTION
**Pull request by joakimriedel**
Fixes #31615

### CODE REVIEW

File: js/tests/unit/modal.js [hunk 1]
```diff
@@ -976,4 +976,42 @@ $(function () {
         backdrop: 'static'
       })
   })
+
+  QUnit.test('should get modal-static class when clicking outside of modal-content if backdrop = static', function (assert) {
```
Comment (Johann-S) [inline js/tests/unit/modal.js:980]: this test already exist here: https://github.com/joakimriedel/bootstrap/blob/joakimriedel/defaultPrevented/js/tests/unit/modal.js#L892

You can improve it by adding your `expect`: `assert.ok($modal.hasClass('modal-static'), 'has modal-static class')`

File: js/tests/unit/modal.js [hunk 2]
```diff
@@ -976,4 +976,42 @@ $(function () {
...
+      })
+  })
+
+  QUnit.test('should not get modal-static class when clicking outside of modal-content if backdrop = static and event is prevented', function (assert) {
+    assert.expect(1)
+    var done = assert.async()
+    var $modal = $('<div class="modal" data-backdrop="static"><div class="modal-dialog" style="transition-duration: 20ms;"/></div>').appendTo('#qunit-fixture')
+
+    $modal.on('hidePrevented.bs.modal', function (e) {
+      e.preventDefault()
```
Comment (Johann-S) [inline js/tests/unit/modal.js:1003]: Adding an assert here you should be good to ensure we pass here

**Review by Johann-S — CHANGES_REQUESTED** [review]
Thanks for your work @joakimriedel 👍 

Just a few modifications and it'll be good to go

File: js/tests/unit/modal.js (hunk 1 above)
Comment (joakimriedel) [reply js/tests/unit/modal.js]: OK, this is done. It required a modification of the old test, adding the transition-duration style. Is that OK?

File: js/tests/unit/modal.js (hunk 2 above)
Comment (joakimriedel) [reply js/tests/unit/modal.js:987]: Not sure what you mean by that?

Comment (Johann-S) [reply js/tests/unit/modal.js:987]: sorry it's due to my bad english 😅 
I think you should add an assert here, to be sure this listener is called

File: js/tests/unit/modal.js (hunk 1 above)
Comment (Johann-S) [reply js/tests/unit/modal.js]: Hmm my bad, I didn't saw the `transition-duration` so you were right it should be better to split those tests

Comment (joakimriedel) [reply js/tests/unit/modal.js:980]: OK, reverted.

File: js/tests/unit/modal.js (hunk 2 above)
Comment (joakimriedel) [reply js/tests/unit/modal.js:1004]: Haha ok got it. Done!

**Comment by joakimriedel** [comment]
Btw @Johann-S I did not commit the dist folder, assumed that is done somehow by CI?

**Comment by Johann-S** [comment]
yep we build the `dist` folder before each release 👌

**Review by Johann-S — APPROVED** [review]

**Pull request merged**